core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -388,24 +409,29 @@ object Checkpoints extends DeltaLogging {
           writeAction()
         }
         Iterator(writtenPath)
-      }.collect().head
+      }.collect()
 
     if (useRename) {
-      val src = new Path(writtenPath)
-      val dest = new Path(path)
-      val fs = dest.getFileSystem(hadoopConf)
       var renameDone = false
       try {
-        if (fs.rename(src, dest)) {
-          renameDone = true
-        } else {
-          // There should be only one writer writing the checkpoint file, so there must be
-          // something wrong here.
-          throw DeltaErrors.failOnCheckpoint(src, dest)
+        writtenPaths.zipWithIndex.foreach { case (writtenPath, index) =>
+          val src = new Path(writtenPath)
+          val dest = new Path(paths(index))
+          val fs = dest.getFileSystem(hadoopConf)
```

Comment:
one last minor comment (sorry. I missed it): We can use `val fs = snapshot.path.getFileSystem(hadoopConf)` outside and then reuse the `fs` object for `rename` and `delete`. This would be faster when people disable file system cache.

Reply:
good point, done